bittide-instances/bin/Hitl.hs

```diff
@@ -0,0 +1,139 @@
+import qualified Bittide.Instances.Hitl.FullMeshHwCc as FullMeshHwCc
+import qualified Bittide.Instances.Hitl.FullMeshSwCc as FullMeshSwCc
+import qualified Bittide.Instances.Hitl.SyncInSyncOut as SyncInSyncOut
+import qualified Bittide.Instances.Hitl.Transceivers as Transceivers
+import qualified Bittide.Instances.Hitl.VexRiscv as VexRiscv
+
+import qualified Bittide.Instances.Hitl.Tcl.ExtraProbes as ExtraProbes
+
+import qualified Data.ByteString.Lazy.Char8 as LazyByteString
+import qualified Clash.Prelude as C
+import qualified Language.Haskell.TH as TH
+
+data Config = Config
+  { name :: String
+  , yaml :: LazyByteString.ByteString
+  }
+
+-- | Known configurations that can be written to @_build/hitl@
+configs :: IO [Config]
+configs = sequence $
+  [ -- Generate config based on Haskell definitions
+    makeConfig 'FincFdec.fincFdecTests                 FincFdec.tests
+  , makeConfig 'FullMeshHwCc.fullMeshHwCcTest          FullMeshHwCc.tests
+  , makeConfig 'FullMeshHwCc.fullMeshHwCcWithRiscvTest FullMeshHwCc.tests
+  , makeConfig 'FullMeshSwCc.fullMeshSwCcTest          FullMeshSwCc.tests
+  , makeConfig 'SyncInSyncOut.syncInSyncOut            SyncInSyncOut.tests
+  , makeConfig 'Transceivers.transceiversUpTest        Transceivers.tests
+  , makeConfig 'VexRiscv.vexRiscvTest                  VexRiscv.tests
+  , makeConfig 'BoardTest.boardTestSimple              BoardTest.testsSimple
+  , makeConfig 'BoardTest.boardTestExtended            BoardTest.testsExtended
+
+    -- Load config from 'bittide-instances/data/test_configs'
+  , loadConfig 'ExtraProbes.extraProbesTest "extraProbesTest.yml"
+  ]
+
+-- | First argument on command line, as Haskell type
+data Arg = WriteConfigs
+  { -- | Fully qualified name of HITL YAML to render. If 'Nothing', render all
+    -- known identifiers
+    fqn :: Maybe String
+  }
+
+-- | Argument parser: currently very boring. Only parses @write-configs@.
+argParser :: Parser Arg
+argParser = subparser $
+  command
+    "write-configs"
+    (info writeConfigsParser (progDesc "Write all known configs to _build/hitl"))
+
+-- | Parser for the write-configs command, now expecting an optional identifier
+writeConfigsParser :: Parser Arg
+writeConfigsParser =
+    fmap WriteConfigs
+  $ optional
+  $ strArgument
+  $    metavar "FULLY_QUALIFIED_NAME"
+    <> help "For example, 'Bittide.Instances.Hitl.FincFdec.fincFdecTests'"
+
+-- | Load config from an existing YAML file in 'data/test_configs'
+loadConfig :: TH.Name -> FilePath -> IO Config
+loadConfig nm fileName = do
+  fullPath <- getDataFileName ("data" </> "test_configs" </> fileName)
+  yamlContents <- LazyByteString.readFile fullPath
+  pure $ Config
+    { name = show nm
+    , yaml = yamlContents
+    }
+
+-- | Create config based on a 'HitlTests'
+makeConfig :: forall a. C.BitPack a => TH.Name -> HitlTests a -> IO Config
+makeConfig nm config = pure $ Config
+  { name = show nm
+  , yaml = packAndEncode config
+  }
```

Comment:
Can we get some output that shows what it did?
Currently we just:
```
[nix-shell:~/bittide-hardware]$ cabal run hitl write-configs
Up to date

```

Reply:
The problems with adding output is that if every command does that, you'll get drowned in uninteresting messages in a large composed system (like Shake). But, like we discussed, I'll add a `-v` flag.